### PR TITLE
Added support for Azure Event Hub

### PR DESCRIPTION
**Pull request by djorgensendk**
This PR adds support in Squadron for Azure Event Hubs. It is possible to create an EventHub namespace with one or more EventHubs in it. Test shows how to publish and consume events on a new event hub and assert that the message published is similar to what this consumed.

### CODE REVIEW

File: src/AzureCloudEventHub.Tests/AzureCloudEventHub.Tests.csproj [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="$(CCTestProjectProps)" Condition="Exists('$(CCTestProjectProps)')" />
+  <PropertyGroup>
+    <RootNamespace>Squadron.AzureCloudEventHub.Tests</RootNamespace>
+    <AssemblyName>Squadron.AzureCloudEventHub.Tests</AssemblyName>
+  </PropertyGroup>
+
+  <ItemGroup>
```
Comment (glucaci) [inline src/AzureCloudEventHub.Tests/AzureCloudEventHub.Tests.csproj:8]: Remove empty itemgroup

File: src/AzureCloudEventHub.Tests/AzureCloudEventHub.Tests.csproj [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
+    </None>
+    <None Update="xunit.runner.json">
+      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
+    </None>
+    <None Update="appsettings.json">
+      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
+    </None>
+  </ItemGroup>
+
```
Comment (glucaci) [inline src/AzureCloudEventHub.Tests/AzureCloudEventHub.Tests.csproj:27]: remove empty line

File: src/AzureCloudEventHub.Tests/AzureNewEventHubResourceTests.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.EventHubs;
+using Xunit;
+
+namespace Squadron.AzureCloudEventHub.Tests
+{
+    public class AzureNewEventHubResourceTests
```
Comment (glucaci) [inline src/AzureCloudEventHub.Tests/AzureNewEventHubResourceTests.cs:10]: Add test also for existing eventhub

File: src/AzureCloudEventHub/AzureCloudEventHub.csproj [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    <PackageReference Include="Microsoft.Azure.Management.EventHub" Version="2.7.2" />
+    <PackageReference Include="System.Collections.Immutable" Version="1.6.0" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\AzureCloud\AzureCloud.csproj" />
+    <ProjectReference Include="..\Core\Core.csproj" />
+  </ItemGroup>
+
+
```
Comment (glucaci) [inline src/AzureCloudEventHub/AzureCloudEventHub.csproj:19]: remove emtpy line

File: src/AzureCloudEventHub/EventHubOptionsBuilder.cs [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        /// Creates a new empty builder
+        /// </summary>
+        /// <returns></returns>
+        public static EventHubOptionsBuilder New() => new EventHubOptionsBuilder();
+
+
+        private EventHubOptionsBuilder()
+            : base()
+        {
+
```
Comment (glucaci) [inline src/AzureCloudEventHub/EventHubOptionsBuilder.cs:25]: remove empty line

File: src/AzureCloudEventHub/EventHubOptionsBuilder.cs [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        private EventHubNamespaceModel _model = new EventHubNamespaceModel();
+        private List<EventHubBuilder> _eventHubs = new List<EventHubBuilder>();
+
+        /// <summary>
+        /// Creates a new empty builder
+        /// </summary>
+        /// <returns></returns>
+        public static EventHubOptionsBuilder New() => new EventHubOptionsBuilder();
+
+
```
Comment (glucaci) [inline src/AzureCloudEventHub/EventHubOptionsBuilder.cs:21]: remove empty line

**Review by glucaci — CHANGES_REQUESTED** [review]

File: src/AzureCloudEventHub/EventHubOptionsBuilder.cs (hunk 6 above)
Comment (djorgensendk) [reply src/AzureCloudEventHub/EventHubOptionsBuilder.cs]: removed

File: src/AzureCloudEventHub/AzureCloudEventHub.csproj (hunk 4 above)
Comment (djorgensendk) [reply src/AzureCloudEventHub/AzureCloudEventHub.csproj]: removed

File: src/AzureCloudEventHub.Tests/AzureNewEventHubResourceTests.cs (hunk 3 above)
Comment (djorgensendk) [reply src/AzureCloudEventHub.Tests/AzureNewEventHubResourceTests.cs:10]: A test has been added in class AzureExistingEventHubResourceTests

File: src/AzureCloudEventHub.Tests/AzureCloudEventHub.Tests.csproj (hunk 2 above)
Comment (djorgensendk) [reply src/AzureCloudEventHub.Tests/AzureCloudEventHub.Tests.csproj:23]: removed

File: src/AzureCloudEventHub.Tests/AzureCloudEventHub.Tests.csproj (hunk 1 above)
Comment (djorgensendk) [reply src/AzureCloudEventHub.Tests/AzureCloudEventHub.Tests.csproj:8]: removed

File: src/AzureCloudEventHub.Tests/appsettings.json [hunk 7]
```diff
@@ -1,13 +1,13 @@
 {
   "Squadron": {
     "Azure": {
-      "SubscriptionId": "",
-      "ResourceGroup": "",
-      "DefaultLocation": "",
+      "SubscriptionId": "3764e1f0-3a1b-4a28-a340-1ed0d8d388fd",
```
Comment (glucaci) [inline src/AzureCloudEventHub.Tests/appsettings.json:4]: Ups... remove these

**Review by glucaci — CHANGES_REQUESTED** [review]

File: src/AzureCloudEventHub.Tests/appsettings.json (hunk 7 above)
Comment (djorgensendk) [reply src/AzureCloudEventHub.Tests/appsettings.json]: thanks - must have gone too fast wehen I committed. All secrets are removed

**Comment by djorgensendk** [comment]
@glucaci it seems like RabbitMQ and ReavenDB tests are failing, which are tests that shouldn't be affected by this PR

**Review by glucaci — APPROVED** [review]

**Pull request merged**